### PR TITLE
PR: Add Contributing Guide based on other Spyder projects

### LINKED ISSUE #66: Add documentation for how to run tests
How is this normally handled? should we add an environment.yaml to set a full environment for qtpy? or should we add a specfile/requirements-test.txt file?

@ccordoba12, @Nodd?

### PR DESCRIPTION
**Pull request by CAM-Gerlach**
Adds a Contributing Guide based on the one for `spyder-docs-sphinx-theme`, which you both previously reviewed and provided a number of helpful suggestions and improvements on (all of which are, of course, incorporated here), and also mirroring where applicable the versions in the website, docs, theme and other repos.

In addition to find-replacing the QtPy project and repo names and eliminating content that isn't relevant, a number of other sections tailored to the specifics of QtPy were added, including:

* A detailed `Running the Tests` section (per #66 )
* A description of installing and selecting Python Qt binding(s) to test against
* An outline of the main branches, and which to use (``master``, aside from critical/security fixes or when otherwise directed)
* More appropriate branch names, commits, etc
* Other clarifications and improvements

Resolve #66 
Part of #85

### CODE REVIEW

**Review by dalthviz — COMMENTED** [review]
Thanks @CAM-Gerlach for the help with this! Left some comments/suggestions, otherwise LGTM :+1:

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+However, this contributing guide should fill you in on most of the basics you need to know.
+
+Let us know if you have any further questions, and we look forward to your contributions!
+
+
+## Reporting Issues
+
+Discover a bug?
+Want a new feature?
+[Open](https://github.com/spyder-ide/qtpy/issues/new/choose) an [issue](https://github.com/spyder-ide/qtpy/issues)!
```
[the remark this answers is not in the document]
Comment (CAM-Gerlach) [reply CONTRIBUTING.md:15]: If and when we add multiple issue template, e.g. bug, feature request, question, etc (as has been the plan for at least the main Spyder repo for years now), the original link will allow users to choose them, while the one you suggested will bypass them (as I confirmed in my testing) and not use our issue template at all, while if we don't have them, the two links work equivalently. This is also the canonical link pointed to by the `New Issue` button. Therefore, to keep things as consistent as practical between projects and reduce the risk of future breakage, both of which impact long-term maintainability, I believe we should keep this as-is.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+### Create and activate a fresh venv
+
+Particularly for development installs, we highly recommend you create and activate a virtual environment to avoid any conflicts with other packages on your system or causing any other issues.
+Of course, you're free to use any environment management tool of your choice (conda, virtualenvwrapper, pyenv, etc).
+
+
+### Create and activate a fresh environment
+
+Particularly for development installs, we highly recommend you create and activate a virtual environment to avoid any conflicts with other packages on your system or causing any other issues.
+To do so with Conda (recommended), simply execute the following:
```
[the remark this answers is not in the document]
Comment (CAM-Gerlach) [reply CONTRIBUTING.md:50]: Yep, absolutely. This was a straight-up copy/paste mistake when editing this, sorry; thanks for the catch! I'll commit this as is.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+Where ``<FEATURE-BRANCH>`` is the name of your feature branch, e.g. ``fix-startup-bug``.
+
+
+
+## Submitting a Pull Request
+
+Finally, create a pull request to the [spyder-ide/qtpy repository](https://github.com/spyder-ide/qtpy/) on Github.
+Make sure to set the target branch to the one you based your PR off of (``master`` or ``X.x``).
+
+Thanks, and we look forward to your generous contributions!
```
[the remark this answers is not in the document]
Comment (CAM-Gerlach) [reply CONTRIBUTING.md:193]: Actually, when working on this originally, I missed pulling in a similar change we all decided on per your and Carlos feedback for the earlier spyder-ide/spyder-docs-sphinx-theme#43 that this is directly based off; here's the wording we agreed on (which I prefer even more here). Sorry I missed that and thanks for the reminder :+1: 

```suggestion
Thanks for taking the time to read and follow this guide, and we look forward to your contributions!
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+git pull upstream <BASE-BRANCH>
+git checkout -b <FEATURE-BRANCH>
+```
+
+Once you've made and tested your changes, commit them with a descriptive message of 74 characters or less written in the imperative tense, with a capitalized first letter and no period at the end.
+For example:
+
+```bash
+git commit -am "Fix bug importing package on Windows"
+```
```
Comment (dalthviz) [inline CONTRIBUTING.md:139]: Should we expand here more about the expected conditions on the commit history to accept a PR @CAM-Gerlach ?

Comment (CAM-Gerlach) [reply CONTRIBUTING.md:140]: Thanks, I added some more tips and suggestions for commit messages and structure, along with a few other things, while also making clear that these are more guidelines than strict rules so not to worry to much about it, and if there is a serious problem we're happy to help (e.g. with a squash or rebase, though pending the release of the reviewer guidelines and your feedback on them I didn't want to set a hard policy there yet).

**Review by dalthviz — APPROVED** [review]
Thanks @CAM-Gerlach !  LGTM :+1:

**Comment by CAM-Gerlach** [comment]
Thanks @dalthviz !

**Pull request merged**